listener/listener.go

```diff
@@ -273,15 +273,16 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 	// Create a handle class that has that function
 	listenerHandle := subReceiver.Listen(ctx, servicebus.HandlerFunc(
 		func(ctx context.Context, msg *servicebus.Message) error {
-			ctx, s := tab.StartSpan(ctx, "go-shuttle.Listener.HandlerFunc")
+			msgCtx, finished := messageContext(ctx, msg)
+			defer finished()
+			msgCtx, s := tab.StartSpan(msgCtx, "go-shuttle.Listener.HandlerFunc")
 			defer s.End()
 			if l.lockRenewalInterval != nil {
-				renewer := peeklock.RenewPeriodically(ctx, *l.lockRenewalInterval, sub, msg)
-				defer renewer.Stop()
+				peeklock.RenewPeriodically(msgCtx, *l.lockRenewalInterval, sub, msg)
 			}
 			currentHandler := handler
 			for !message.IsDone(currentHandler) {
```

Comment:
Not sure if this is a good idea or not, but do we possibly want to short circuit the operation here if msgCtx.Done() is true?

Any future calls to the handler would be pointless, correct?